os_stub/spdm_device_secret_lib_sample/cert.c

```diff
@@ -461,8 +481,13 @@ boolean read_responder_root_public_certificate_by_size(
     cert_chain->length = (uint16_t)cert_chain_size;
     cert_chain->reserved = 0;
 
-    spdm_hash_all(base_hash_algo, file_data, file_size,
+    res = spdm_hash_all(base_hash_algo, file_data, file_size,
               (uint8_t *)(cert_chain + 1));
+    if (!res) {
+        free(file_data);
+        free(cert_chain);
+        return res;
+    }

```

Comment:
Some of these changes will go away once spdm_hash_all fixes are merged.
Was just difficult to develop both without ensuring there won't be merge conflicts later